src/Framework/test/TestData.cs

```diff
@@ -154,6 +155,12 @@ static TestData()
                 "System.Threading.RateLimiting",
             };
 
+        // System.Diagnostics.EventLog.Messages is only present in the Windows build.
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            ListedSharedFxAssemblies.Remove("System.Diagnostics.EventLog.Messages");
```

Comment:
This won't work on Helix since we always run tests against the windows assets there (we don't intend to change that) - you could add `!SkipOnHelixAttribute.OnHelix()` to this condition to account for that

Reply:
I'll make the change.
I'm curious, is there a reason you only run these tests against the Windows assets?

Reply:
We try to run every test we can on Helix because it saves resources, and helix works by building assets once & sending them off to multiple test queues. So the answer is essentially "because we can" - but, I might be swayed by an argument that these tests should _not_ be run on Helix because they're testing if the assets were _produced_ correctly on the target OS, not that they _run_ correctly there. @dougbu what do you think?

Reply:
My point is that these tests check "Was the sharedFx built correctly?". Since we run them in helix, we're only testing "Was the sharedFx built on windows built correctly?" - we don't run this test against a SharedFx built on Linux or Mac (in CI).

Reply:
> The condition now looks like it'll work fine when run on Helix agents. Am I missing something❔

The tests failed on non-windows on helix in the first iteration because of this test, because of this condition (they ran on non-windows, but tested against a SharedFx built on windows)

Reply:
> My point is that these tests check "Was the sharedFx built correctly?". Since we run them in helix, we're only testing "Was the sharedFx built on windows built correctly?" - we don't run this test against a SharedFx built on Linux or Mac (in CI).

One way to add such testing would be to move the tests off Helix. Unfortunately, that goes against our preference to test _everything_ we can in Helix and avoid duplicate testing.

This also pretty normal because _all_ of our assemblies are built on Windows and tested on the Helix agents. The main unusual thing is that we actually ship shared Fx packages built on non-Windows agents.

Until and unless we centralize `crossgen2` execution to the Windows build, let's ignore this. Thoughts❔

Reply:
> > The condition now looks like it'll work fine when run on Helix agents. Am I missing something❔
> 
> The tests failed on non-windows on helix in the first iteration because of this test, because of this condition (they ran on non-windows, but tested against a SharedFx built on windows)

Are we 🆗 from your perspective here now❔

Reply:
> One way to add such testing would be to move the tests off Helix. Unfortunately, that goes against our preference to test everything we can in Helix and avoid duplicate testing. This also pretty normal because all of our assemblies are built on Windows and tested on the Helix agents. The main unusual thing is that we actually ship shared Fx packages built on non-Windows agents.

There are some exceptions to that rule though, and I think this should be one of them - these tests are different from others in that they're not testing "do these bits run on this platform", they're testing "Did we produce the correct set of bits for this platform". As it is today, there's no point in running this test on the Windows, Linux, and MacOS queues - all three are testing "Did we build the correct set of bits for windows", and none are testing "Did we build the correct set of bits for Mac/Linux"

Reply:
> There are some exceptions to that rule though, and I think this should be one of them

The exceptions are all **bugs** and I would prefer we don't pile on more technical debt. But I also don't disagree our tests of the shared Fx packages is a touch of tunnel vision.

My thought was also about urgency because we haven't hit significant platform-specific problems w/ the shared Fx packages (ignoring those that show up when building in the CI) more than a couple of times (if that). I also plan to try centralizing `crossgen2` sooner rather than later.

Reply:
I don't think we need to block this PR on it - I opened https://github.com/dotnet/aspnetcore/issues/46620 to track fixing it in the future.